graphql/util.go

```diff
@@ -1163,3 +1165,65 @@ func interfaceToMap(ctx context.Context, data interface{}) (map[string]interface
 
 	return mapField, nil
 }
+
+type HasMatchingTasks struct {
+	Id       string
+	HasTasks bool
+}
+
+func concurrentlyBuildHasMatchingTasksMap(ctx context.Context, versions []model.Version, opts task.HasMatchingTasksOptions, hasMatchingTasksMap map[string]bool) error {
+	wg := sync.WaitGroup{}
+	input := make(chan model.Version, len(versions))
+	output := make(chan HasMatchingTasks, len(versions))
+	catcher := grip.NewBasicCatcher()
+
+	for _, v := range versions {
+		input <- v
+	}
+	close(input)
+
+	// Limit number of parallel requests to the DB.
+	const maxParallel = 20
+	workers := util.Min(maxParallel, len(versions))
+	for i := 0; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for i := range input {
+				hasTasks, err := task.HasMatchingTasks(ctx, i.Id, opts)
+				if err != nil {
+					catcher.Add(err)
+					continue
+				}
+				output <- HasMatchingTasks{Id: i.Id, HasTasks: hasTasks}
+			}
+		}()
+	}
+	wg.Wait()
+	close(output)
+
+	if catcher.HasErrors() {
+		return werrors.Wrap(catcher.Resolve(), "finding matching tasks")
```

Comment:
`werrors` is the same as `errors`, so I think this is incorrectly imported twice.